### PR TITLE
Add pitch for y dimension in cpu copy

### PR DESCRIPTION
**Pull request by erikzenker**
Fixes the copy bug mentioned in #138 for the cpu accelerators.
The copy algorithm used wrong pitches to "jump" into the next z slice.
Y pitch was added as member and used for next z slice calculation.

### CODE REVIEW

**Comment by erikzenker** [comment]
A unit test, verifying various kinds of copies, should follow.

**Comment by BenjaminW3** [comment]
Thanks for fixing this!
I would prefer to have unit tests on a per concept implementation base. E.g., the `ViewSubViewTest` I have just added should be extended to test its copy and set implementations. All of the tests then should be done equally for `BufCpu`, `BufGpu` and `ViewPlainPtr`.

File: include/alpaka/mem/buf/cpu/Copy.hpp [hunk 1]
```diff
@@ -97,8 +97,10 @@ namespace alpaka
                                 m_dstDepth(static_cast<Size>(extent::getDepth(viewDst))),
                                 m_srcDepth(static_cast<Size>(extent::getDepth(viewSrc))),
 #endif
-                                m_dstPitchBytes(static_cast<Size>(mem::view::getPitchBytes<dim::Dim<TViewDst>::value - 1u>(viewDst))),
-                                m_srcPitchBytes(static_cast<Size>(mem::view::getPitchBytes<dim::Dim<TViewSrc>::value - 1u>(viewSrc))),
+                                m_dstPitchBytesX(static_cast<Size>(mem::view::getPitchBytes<dim::Dim<TViewDst>::value - 1u>(viewDst))),
+                                m_srcPitchBytesX(static_cast<Size>(mem::view::getPitchBytes<dim::Dim<TViewSrc>::value - 1u>(viewSrc))),
+                                m_dstPitchBytesY(static_cast<Size>(mem::view::getPitchBytes<dim::Dim<TViewDst>::value - 2u>(viewDst))),
```
Comment (BenjaminW3) [inline include/alpaka/mem/buf/cpu/Copy.hpp]: Currently you are failing here beause for a 1D Buffer this is asking for the Pitch of the dimension -1.
To fix the current bug, it should be enough to specialize a copy for the 1D case.

Comment (psychocoderHPC) [inline include/alpaka/mem/buf/cpu/Copy.hpp]: dim::Dim<TViewDst>::value
Maybe we can write this to solve the 1D problem
```C++
constexpr dim = dim::Dim<TViewDst>::value;
m_dstPitchBytesY(static_cast<Size>(
    mem::view::getPitchBytes<
    dim::Dim<
            TViewDst
        >::value - ( 2u % dim )
    >(
        viewDst
    )
)
```

**Comment by erikzenker** [comment]
Fixed :-)

**Comment by BenjaminW3** [comment]
Could you make this a single commit, please?

**Comment by erikzenker** [comment]
I did a rebase against the current develop branch

**Pull request merged**